mongo/options/clientoptions.go

```diff
@@ -74,7 +74,9 @@ type ContextDialer interface {
 // Password: the password for authentication. This must not be specified for X509 and is optional for GSSAPI
 // authentication.
 //
-// PasswordSet specifies if the password is actually set, since an empty password is a valid password.
+// PasswordSet: For GSSAPI, this must be true if a password is specified, even if the password is the empty string, and
```

Comment:
I was concerned about the `context of the running process` bit, but I talked to Craig and he said actually clarifying all of the details around that would require too much explanation without much benefit. Happy to update it if anyone has any suggestions for rephrasing it, though.

Reply:
So you need to set this to true for GSSAPI even if the password in a non-empty string?

Reply:
Yes.

Reply:
That seems silly, but it means this is accurate.